octobot/strategy_optimizer/strategy_design_optimizer.py

```diff
@@ -0,0 +1,834 @@
+                                                                       tentacle_class)
+            dict_util.nested_update_dict(current_config, updated_values)
+            tentacles_manager_api.update_tentacle_config(local_tentacles_setup_config, tentacle_class, current_config)
+        return local_tentacles_setup_config
+
+    @staticmethod
+    def _ensure_local_config_folders(run_folder):
+        if not os.path.exists(run_folder):
+            os.makedirs(run_folder)
+        tentacles_specific_config_folder = os.path.join(run_folder,
+                                                        tentacles_manager_constants.TENTACLES_SPECIFIC_CONFIG_FOLDER)
+        if not os.path.exists(tentacles_specific_config_folder):
+            os.mkdir(tentacles_specific_config_folder)
+
+    async def _generate_and_store_backtesting_runs_schedule(self):
+        runs = self._generate_runs()
+        await self._save_run_schedule(runs)
+        return runs
+
+    def _generate_runs(self):
+        iterations = [i for i in self._get_config_possible_iterations() if i]
+        runs = {
+            index: run
+            for index, run in enumerate(itertools.product(*iterations))
+            if self._is_run_allowed(run)
+        }
+        if runs:
+            for run in runs.values():
+                for run_input in run:
+                    # do not store self.CONFIG_KEY
+                    run_input.pop(self.CONFIG_KEY, None)
+            return runs
+        raise RuntimeError("No optimizer run to schedule with this configuration")
+
+    def _is_run_allowed(self, run):
+        for run_filter_config in self.optimizer_config[self.CONFIG_FILTER_SETTINGS]:
+            if self._is_filtered(run, run_filter_config):
+                return False
+        return True
+
+    def _is_filtered(self, run, run_filter_config):
+        left_operand, operator, right_operand = self._parse_filter_entry(run, run_filter_config)
+        if not (left_operand and right_operand):
+            return False
+        try:
+            left_operand = decimal.Decimal(left_operand)
+        except decimal.InvalidOperation:
+            left_operand = str(left_operand)
+        try:
+            right_operand = decimal.Decimal(right_operand)
+        except decimal.InvalidOperation:
+            right_operand = str(right_operand)
+
+        if operator == "lower_than":
+            return left_operand < right_operand
+        if operator == "higher_than":
+            return left_operand > right_operand
+        if operator == "lower_or_equal_to":
+            return left_operand <= right_operand
+        if operator == "higher_or_equal_to":
+            return left_operand >= right_operand
+        if operator == "equal_to":
+            return left_operand == right_operand
+        if operator == "different_from":
+            return left_operand != right_operand
+        raise RuntimeError(f"Unknown operator: {operator}")
```

Comment:
I think we should move this to commons, we may need a similar feature somewhere else